File: JavaImpl/src/main/java/org/example/Chapter2_Selection_Sort/ArrayExample.java
package org.example.Chapter2_Selection_Sort;


import java.util.ArrayList;
import java.util.Arrays;

public class ArrayExample {
    public static void main(String[] args) {
        // Creating an array
        int[] numbers = {1, 2, 3, 4, 5};

        // Reading elements
        for (int i = 0; i < numbers.length; i++) {
            System.out.println("Element at index " + i + ": " + numbers[i]);
        }

        // Inserting an element at the end (in case of dynamic array)
        Integer[] integerArray = Arrays.stream(numbers).boxed().toArray(Integer[]::new);
        ArrayList<Integer> list = new ArrayList<>(Arrays.asList(integerArray));
        list.add(6); // Adding an element at the end
        System.out.println("Elements after addition: " + list);
    }
}

File: KotlinImpl/src/main/kotlin/Chapter2_Selection_Sort/ArrayExample.kt
package Chapter2_Selection_Sort

fun main() {
    // Creating an array
    val numbers = arrayOf(1, 2, 3, 4, 5)

    // Reading elements
    for (i in numbers.indices) {
        println("Element at index $i: ${numbers[i]}")
    }

    // Inserting an element at the end
    val list = numbers.toMutableList() // Convert array to a mutable list
    list.add(6) // Adding an element at the end
    println("Elements after addition: $list")
}
